modules/client/src/main/java/org/apache/ignite/internal/client/tx/ClientTransaction.java

```diff
@@ -134,6 +153,23 @@ public static ClientTransaction get(Transaction tx) {
                     + "'. Use IgniteClient.transactions() to start transactions.");
         }
 
-        return (ClientTransaction) tx;
+        ClientTransaction clientTx = (ClientTransaction) tx;
+
+        var state = clientTx.state.get();
+
+        if (state == STATE_OPEN) {
+            return clientTx;
+        }
+
+        // Match org.apache.ignite.internal.tx.TxState enum:
+        String stateStr = state == STATE_COMMITTED ? "COMMITTED" : "ABORTED";
+
+        throw new TransactionException(
```

Comment:
We have tests that run against client and embedded APIs and ensure consistent behavior (ItSqlSynchronousApiTest & ItSqlClientSynchronousApiTest, etc).

It would be great to rely on the server only for this, but when the tx is finished, the server removes it from `ClientResourceRegistry` and we can't use it anymore from the client.